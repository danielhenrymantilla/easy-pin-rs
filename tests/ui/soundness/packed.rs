#![feature(trivial_bounds)]

#![allow(unused)]

use ::easy_pin::{
    easy_pin,
    PinSensitive,
};
use ::core::ptr::NonNull;

#[repr(packed)]
#[easy_pin]
pub
struct SelfReferential {
    #[transitively_pinned]
    string: PinSensitive<String>,
}

fn main ()
{}
