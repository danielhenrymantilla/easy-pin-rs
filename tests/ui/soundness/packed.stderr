error: borrow of packed field is unsafe and requires unsafe function or block (error E0133)
  --> $DIR/packed.rs:15:5
   |
15 | /     #[transitively_pinned]
16 | |     string: PinSensitive<String>,
   | |__________^
   |
note: lint level defined here
  --> $DIR/packed.rs:15:5
   |
15 |     #[transitively_pinned]
   |     ^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #46043 <https://github.com/rust-lang/rust/issues/46043>
   = note: fields of packed structs might be misaligned: dereferencing a misaligned pointer or even just creating a misaligned reference is undefined behavior
